firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3319,18 +3319,40 @@ class BrowserViewController: UIViewController,
         if let url {
             switchToTabForURLOrOpen(url, isPrivate: isPrivate)
         } else {
-            guard let selectedTab = tabManager.selectedTab else { return }
-            if selectedTab.isPrivate == isPrivate, selectedTab.isFxHomeTab {
-                focusLocationTextField(forTab: selectedTab)
-            } else {
-                openBlankNewTab(
-                    focusLocationField: options?.contains(.focusLocationField) == true,
-                    isPrivate: isPrivate
-                )
+            let isFocusLocationTextFieldOption = options?.contains(.focusLocationField) == true
+
+            // Avoid race condition; if we're restoring tabs, wait to process URL until completed. [FXIOS-10916]
+            // Wait for tabsRestoration because we need selectedTab
+            // selectedTab is nil when open FF from widget
```

Comment:
```suggestion
            // Wait for tabs restoration because we need the `selectedTab`.
            // The `selectedTab` is `nil` when open firefox from a widget.
```